packages/core/src/data-module/IotAppKitDataModule.ts

```diff
@@ -88,22 +88,39 @@ export class IotAppKitDataModule implements DataModule {
 
     const requests = requiredStreams
       .map(({ resolution, id, cacheSettings }) => {
+        const streamCacheSettings = { ...this.cacheSettings, ...cacheSettings };
+
         const dateRanges = getDateRangesToRequest({
           store: this.dataCache.getState(),
           start: viewportStartDate(viewport),
           end: viewportEndDate(viewport),
           resolution: parseDuration(resolution),
           dataStreamId: id,
-          cacheSettings: { ...this.cacheSettings, ...cacheSettings },
+          cacheSettings: streamCacheSettings,
         });
 
         return {
           dateRanges,
-          request: { id, resolution },
+          request: { id, resolution, cacheSettings: streamCacheSettings },
         };
       })
       .flatMap(({ dateRanges, request }) =>
-        dateRanges.map(([rangeStart, rangeEnd]) => ({ start: rangeStart, end: rangeEnd, ...request }))
+        dateRanges.map(([rangeStart, rangeEnd]) => ({
+          start: rangeStart,
+          end: rangeEnd,
+
+          // determine if the most recent point before start is already cached
+          // to avoid redundant fetchMostRecentBeforeStart
+          leadingPointInCache: checkCacheForRecentPoint({
```

Comment:
I'm not convinced we need to introduce this new concept and carry it through out. We already have a function that returns the date ranges the request settings to request, what if we instead moved all the caching logic in one place, `getDateRangesToRequest`?

Reply:
The trouble with fetching leading points is that for a potentially large date range, we only want to fetch a single point (whereas currently, all data within these date ranges are fetched). Request settings don't have the ability to differentiate between different _types_ of requests so this new request setting introduces that capability.